Source/WebCore/platform/graphics/cocoa/IOSurface.mm

```diff
@@ -348,7 +348,10 @@ static IntSize computeMaximumSurfaceSize()
 #else
     UNUSED_PARAM(hostWindow);
 #endif
-
+#if HAVE(CGCONTEXT_SET_OWNER_IDENTITY)
+    if (m_resourceOwner)
+        CGContextSetOwnerIdentity(m_cgContext.get(), m_resourceOwner.taskIdToken());
+#endif
```

Comment:
Is Metal used for IOSurface context based only? There are many cases where we create unaccelerated ImageBuffers (CGBitmapContext based ImageBuffers). Do we need to attribute the resources of these contexts to WebProcess as well?

Also should not GraphicsContext be the place we add this call to? So ensureGraphicsContext can look like this:

```
GraphicsContext& IOSurface::ensureGraphicsContext()
{
    if (m_graphicsContext)
        return *m_graphicsContext;

    m_graphicsContext = makeUnique<GraphicsContextCG>(ensurePlatformContext());
    m_graphicsContext->setIsAcceleratedContext(true);
    m_graphicsContext->setOwnerIdentity(m_resourceOwner);

    return *m_graphicsContext;
}   
```
And GraphicsContextCG::setOwnerIdentity() can have the `#if HAVE(CGCONTEXT_SET_OWNER_IDENTITY)` and the call to the SPI `CGContextSetOwnerIdentity()`.

Reply:
> Is Metal used for IOSurface context based only?

Yes, Metal is IOSurface only.

> There are many cases where we create unaccelerated ImageBuffers (CGBitmapContext based ImageBuffers). Do we need to attribute the resources of these contexts to WebProcess as well?

Later on, we may need to do that. Currently there is no mechanism to implement that, there is no implementation for it and we don't have data on what we would gain doing that.

> Also should not GraphicsContext be the place we add this call to? So ensureGraphicsContext can look like this:
>   m_graphicsContext->setOwnerIdentity(m_resourceOwner);

This would be the ideal, but given how the code is organised, this would look inconsistent and be incorrect.
From the code organisation perspective:
The CGContext is not owned by the GraphicsContextCG, and as such setting a persistent owner identity in GraphicsContextCG is inconsistent. When the CGContext is used for some other purpose (since it's not owned for this purpose), nothing guarantees that the owner identity is correctly setup. The owner identity cannot be unset.

From correctness perspective, we do not always have a GraphicsContextCG when doing operations on the CGContext. As such it's not ideal that we wouldn't use the attribution for the cases where we don't use  GraphicsContextCG.

The current code structure is such that the caller of GraphicsContextCG configures the underlying CGContext and then creates GraphicsContextCG. This patch adds to the configuration step.